build/Targets/Settings.props

```diff
@@ -90,12 +101,42 @@
 
     <!-- https://github.com/dotnet/roslyn/issues/21272 -->
     <GenerateResourceMSBuildRuntime Condition="'$(TargetFramework)' == 'netcoreapp2.0'">CurrentRuntime</GenerateResourceMSBuildRuntime>
+   
+    <!-- 
+      Official build:
+       - Build standalone Portable PDBs to reduce the size of the binaries.
+       - Convert Portable PDBs to Windows PDBs and publish the converted PDBs to Symbol Store to allow WinDBG, 
+         Watson and other tools to find symbol format they understand.
+        
+      Jenkins build:
+       - Embed PDBs to make it easier to debug Jenkins crash dumps.
+       
+      Developer build:
+       - Embed PDBs to be consistent with Jenkins builds.
+    -->
+    <RoslynDebugType Condition="'$(OfficialBuild)' == 'true'">portable</RoslynDebugType>
+    <RoslynDebugType Condition="'$(OfficialBuild)' != 'true'">embedded</RoslynDebugType>
+
+    <!-- 
+      The source root path used for deterministic normalization of source paths.
+      
+      If set the value is used in PathMap to instruct the compiler to normalize source paths 
+      emitted to the binary and PDB and as a root for Source Link mapping.
+      If set must end with a directory separator.
+      
+      If unset the PathMap need to be set explicitly to facilitate full determinism.
+      
+      The path can't be just a single '/' due to bug in Roslyn compilers:
+      https://github.com/dotnet/roslyn/issues/22815
+    -->
+    <DeterministicSourceRoot Condition="'$(DeveloperBuild)' != 'true'">/_/</DeterministicSourceRoot>
```

Comment:
Will this break anything once we fix the bug and then set it to /?

Reply:
It should work well.

---
In reply to: [147290830](https://github.com/dotnet/roslyn/pull/22850#discussion_r147290830) [](ancestors = 147290830)